tests/integration/targets/latest/tasks/object_permission.yml

```diff
@@ -0,0 +1,89 @@
+---
+##
+##
+### PYNAUTOBOT_OBJECT_PERMISSIONS
+##
+##
+- name: "PERMISSION 1: Necessary info creation"
+  networktocode.nautobot.object_permission:
+    url: "{{ nautobot_url }}"
+    token: "{{ nautobot_token }}"
+    name: Test PERMISSION One
+    enabled: true
+    object_types:
+      - "dcim.device"
+    actions:
+      - view
+    state: present
+  register: test_one
+
+- name: "PERMISSION 1: ASSERT - Necessary info creation"
+  assert:
+    that:
+      - test_one is changed
+      - test_one['diff']['before']['state'] == "absent"
+      - test_one['diff']['after']['state'] == "present"
+      - test_one['permission']['name'] == "Test PERMISSION One"
+      - test_one['msg'] == "permission Test PERMISSION One created"
+
+- name: "PERMISSION 2: Create duplicate"
+  networktocode.nautobot.object_permission:
+    url: "{{ nautobot_url }}"
+    token: "{{ nautobot_token }}"
+    name: Test PERMISSION One
+    enabled: true
+    actions:
+      - view
+    object_types:
+      - "dcim.device"
+    state: present
+  register: test_two
+
+- name: "PERMISSION 2: ASSERT - Create duplicate"
+  assert:
+    that:
+      - not test_two['changed']
+      - test_two['permission']['name'] == "Test PERMISSION One"
+      - test_two['msg'] == "permission Test PERMISSION One already exists"
+
+- name: "PERMISSION 3: ASSERT - Update"
+  networktocode.nautobot.object_permission:
+    url: "{{ nautobot_url }}"
+    token: "{{ nautobot_token }}"
+    name: "Test PERMISSION One"
+    enabled: true
+    actions:
+      - add
+    object_types:
+      - "dcim.device"
+    state: present
+  register: test_three
+
+- name: "PERMISSION 3: ASSERT - Updated"
+  assert:
+    that:
+      - test_three is changed
+      - test_three['diff']['after']['actions'] == ["add"]
+      - test_three['permission']['name'] == "Test PERMISSION One"
+      - test_three['permission']['actions'] == ["add"]
+      - test_three['msg'] == "permission Test PERMISSION One updated"
+
+- name: "PERMISSION 4: ASSERT - Delete"
+  networktocode.nautobot.object_permission:
+    url: "{{ nautobot_url }}"
+    token: "{{ nautobot_token }}"
+    name: "Test PERMISSION One"
+    enabled: true
+    actions:
+      - view
+    object_types:
+      - "dcim.device"
+    state: absent
+  register: test_four
+
+- name: "PERMISSION 4: ASSERT - Delete"
+  assert:
+    that:
+      - test_four is changed
+      - test_four['permission']['name'] == "Test PERMISSION One"
+      - test_four['msg'] == "permission Test PERMISSION One deleted"
```

Comment:
Can you please change tests to the pattern of 2x create, 2x update and 2x delete: first to make the change and second to verify idempotency? You can skip the 2x update on group as you know it's only 1 parameter.

Reply:
think this is done